src/main/java/org/javarosa/core/model/instance/DataInstance.java

```diff
@@ -289,7 +289,11 @@ public void setCacheHost(CacheHost cacheHost) {
         this.mCacheHost = cacheHost;
     }
 
+    /**
+     * Cleans reference caches maintained by the instance and TreeElements contained in the contextNode
+     */
     public void cleanCache() {
         referenceCache.clear();
+        getBase().clearVolatiles();

```

Comment:
Just to clarify, this `cleanCache` function is already being called in the correct way and we're just adding more cleanup code. Is that correct?

Reply:
yes, for this particular use case of deleting repeat it gets called [here](https://github.com/dimagi/commcare-core/blob/master/src/main/java/org/javarosa/core/model/FormDef.java#L376)